macOSでのdateコマンド対応を追加

Kosuke-Ito opened the pull request:
## 概要
Mac OS で使用した際に、macOS の date は GNU date とは異なるため、 -d オプションは使えないのでMac OSの場合では別の方法で対応したいです

- エラーログ
```bash
❯ gh furik      
date: illegal option -- d
usage: date [-jnRu] [-I[date|hours|minutes|seconds|ns]] [-f input_fmt]
            [ -z output_zone ] [-r filename|seconds] [-v[+|-]val[y|m|w|d|H|M|S]]
            [[[[mm]dd]HH]MM[[cc]yy][.SS] | new_date] [+output_fmt]
```

## 対応
使用しているdateコマンドの種類で判定します

## 動作確認
### macOS環境（BSD date）での検証
```
✓ BSD dateとして正しく判定
✓ FROM: 2025-12-11T00:00:00Z
✓ フォーマット検証: OK
✓ GNU date構文は使えない（期待通り）
```

### Linux環境（GNU date）での検証
```
✓ GNU dateとして正しく判定
✓ FROM: 2025-12-11T00:00:00Z
✓ フォーマット検証: OK
✓ BSD date構文は使えない（期待通り）
```

Kosuke-Ito committed:
fix: macOSでのdateコマンド対応を追加

macOS (BSD date) とLinux (GNU date) で異なるdateコマンドの構文に対応するため、プラットフォーム判定を追加して適切なコマンドを使用するように修正

Kosuke-Ito committed:
不要なコメントを削除

kenchan commented on gh-furik:
```diff
@@ -1,7 +1,13 @@
 set -e
 
-FROM=$(date -d "yesterday" +%Y-%m-%dT00:00:00Z)
+if [[ "$OSTYPE" == "darwin"* ]]; then
```
MacOSでもGNU dateを使っているケースはありえるので以下のような判定でうまくいくか試してほしいです！

```
if date --version >/dev/null 2>&1; then
    echo "これは GNU date です"
else
    echo "これは BSD date です"
fi
```

Kosuke-Ito committed:
fix: dateコマンド判定をdate --versionに変更

レビューコメント対応: OSTYPEによる判定から、実際のdateコマンドの種類を判定する方法に変更。
macOSでGNU dateを使っている環境で動作するように改善。

kenchan commented:
Thunks！
